src/dayview/nowview.ts

```diff
@@ -0,0 +1,202 @@
+    .append("svg")
+    .attr("class", "stateArea")
+    .attr("width", stateWidth)
+    .attr("height", dataHeight)
+    .attr("x", dataWidth);
+
+  // Get current state data using allFieldsSvg with a tiny time window
+  const stateEndUtc = viewportEnd.toUTC().toISO();
+  const stateStartUtc = viewportEnd.minus({ milliseconds: 1 }).toUTC().toISO(); // Small window to get current state
+
+  const stateFieldsSvg = await allFieldsSvg({
+    db,
+    startUtc: stateStartUtc,
+    endUtc: stateEndUtc,
+    width: stateWidth,
+    height: dataHeight,
+  });
+
+  if (stateFieldsSvg) {
+    stateArea.append(() => stateFieldsSvg);
+  }
+
+  // Add border around state panel
+  stateArea
+    .append("rect")
+    .attr("x", 0)
+    .attr("y", 0)
+    .attr("width", stateWidth)
+    .attr("height", dataHeight)
+    .attr("fill", "none")
+    .attr("stroke", "#ffcc00")
+    .attr("stroke-width", 2)
+    .attr("stroke-opacity", 0.7);
+
+  // Create time scale with current time at right edge of timeline
+  const timeScale = d3
+    .scaleTime()
+    .domain([viewportStart.toJSDate(), viewportEnd.toJSDate()])
+    .range([0, dataWidth]);
+
+  // Add a vertical line indicating the current time
+  plot
+    .append("line")
+    .attr("class", "current-time-line")
+    .attr("x1", timeScale(viewportEnd.toJSDate()))
+    .attr("x2", timeScale(viewportEnd.toJSDate()))
+    .attr("y1", 0)
+    .attr("y2", dataHeight)
+    .attr("stroke", "#ffcc00")
+    .attr("stroke-width", 2)
+    .attr("stroke-dasharray", "4,2");
+
+  const tickValues = [
+    viewportEnd.toJSDate(),
+    viewportEnd.minus({ minutes: 5 }).toJSDate(),
+    viewportEnd.minus({ minutes: 10 }).toJSDate(),
+    viewportEnd.minus({ minutes: 15 }).toJSDate(),
+  ];
+
+  const timeAxis = plot
+    .append("g")
+    .attr("transform", `translate(0, ${dataHeight})`)
+    .call(
+      d3
+        .axisBottom(timeScale)
+        .tickValues(tickValues)
+        .tickFormat((d) => {
+          const date = d as Date;
+          const diffMinutes = Math.round(
+            (date.getTime() - viewportEnd.toJSDate().getTime()) / 60000,
+          );
+          return diffMinutes === 0
+            ? `${viewportEnd.hour}:${viewportEnd.minute.toString().padStart(2, "0")}`
+            : `${diffMinutes}m`;
+        }),
+    );
+
+  timeAxis.selectAll("text").attr("fill", "white");
+  timeAxis.selectAll("line").attr("stroke", "white");
+  timeAxis.selectAll("path").attr("stroke", "white");
+
+  // Add vertical grid lines at the tick positions
+  plot
+    .append("g")
+    .selectAll(".gridline")
+    .data(tickValues)
+    .enter()
+    .append("line")
+    .attr("class", "gridline")
+    .attr("x1", (d) => timeScale(d))
+    .attr("x2", (d) => timeScale(d))
+    .attr("y1", 0)
+    .attr("y2", dataHeight)
+    .attr("stroke", "white")
+    .attr("stroke-opacity", 0.4);
+
+  // Add keyboard navigation script
+  const script = document.createElement("script");
+  script.textContent = getNowviewScript();
```

Comment:
Adding the script seems to break the .svg files (because it is for html). I'm going to remove the script for now anyway--but this is an important thing to consider for later

Reply:
Noted